include/mbgl/style/style_properties.hpp

```diff
@@ -169,22 +165,16 @@ class BackgroundPaintProperties {
     float opacity = 1.0f;
     Color color = {{ 0, 0, 0, 1 }};
     Faded<std::string> image;
+
+    inline bool isVisible() const {
+        return opacity > 0;
+    }
 };
```

Comment:
We'd want to do that only if we actually needed runtime polymorphism from `PaintProperties`. So far I don't see the need -- everywhere `isVisible()` is currently used, it's with a concrete type.